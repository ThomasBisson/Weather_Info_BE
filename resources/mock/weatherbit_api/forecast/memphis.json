[
    {
        "app_max_temp": 13.8,
        "app_min_temp": 8.3, 
        "clouds": 48,        
        "clouds_hi": 51,     
        "clouds_low": 50,    
        "clouds_mid": 9,
        "datetime": "2023-12-03",
        "dewpt": 8.1,
        "high_temp": 14.5,
        "low_temp": 5.9,
        "max_dhi": null,
        "max_temp": 14.5,
        "min_temp": 7.9,
        "moon_phase": 0.545474,
        "moon_phase_lunation": 0.7,
        "moonrise_ts": 1701661359,
        "moonset_ts": 1701627782,
        "ozone": 295.3,
        "pop": 0,
        "precip": 0,
        "pres": 1004.6,
        "rh": 87,
        "slp": 1014.5,
        "snow": 0,
        "snow_depth": 0,
        "sunrise_ts": 1701607854,
        "sunset_ts": 1701643704,
        "temp": 10.3,
        "ts": 1701583260,
        "uv": 2.9,
        "valid_date": "2023-12-03",
        "vis": 10.674,
        "weather": {
            "code": 803,
            "icon": "c03d",
            "description": "Broken clouds"
        },
        "wind_cdir": "SSW",
        "wind_cdir_full": "south-southwest",
        "wind_dir": 210,
        "wind_gust_spd": 5.1,
        "wind_spd": 3.4
    },
    {
        "app_max_temp": 10.5,
        "app_min_temp": 2.7,
        "clouds": 27,
        "clouds_hi": 0,
        "clouds_low": 39,
        "clouds_mid": 18,
        "datetime": "2023-12-04",
        "dewpt": 5,
        "high_temp": 11.1,
        "low_temp": 2.6,
        "max_dhi": null,
        "max_temp": 11.1,
        "min_temp": 3.9,
        "moon_phase": 0.45003,
        "moon_phase_lunation": 0.73,
        "moonrise_ts": 1701751286,
        "moonset_ts": 1701715623,
        "ozone": 316.2,
        "pop": 0,
        "precip": 0,
        "pres": 1008.7,
        "rh": 81,
        "slp": 1018.7,
        "snow": 0,
        "snow_depth": 0,
        "sunrise_ts": 1701694305,
        "sunset_ts": 1701730103,
        "temp": 8.2,
        "ts": 1701669660,
        "uv": 1.6,
        "valid_date": "2023-12-04",
        "vis": 18.141,
        "weather": {
            "code": 802,
            "icon": "c02d",
            "description": "Scattered clouds"
        },
        "wind_cdir": "W",
        "wind_cdir_full": "west",
        "wind_dir": 276,
        "wind_gust_spd": 6.9,
        "wind_spd": 4.5
    },
    {
        "app_max_temp": 13.3,
        "app_min_temp": 0.2,
        "clouds": 6,
        "clouds_hi": 2,
        "clouds_low": 4,
        "clouds_mid": 2,
        "datetime": "2023-12-05",
        "dewpt": 2.3,
        "high_temp": 13.8,
        "low_temp": 0.9,
        "max_dhi": null,
        "max_temp": 13.8,
        "min_temp": 2.6,
        "moon_phase": 0.355754,
        "moon_phase_lunation": 0.76,
        "moonrise_ts": 1701841115,
        "moonset_ts": 1701803400,
        "ozone": 328.2,
        "pop": 0,
        "precip": 0,
        "pres": 1011,
        "rh": 72,
        "slp": 1021,
        "snow": 0,
        "snow_depth": 0,
        "sunrise_ts": 1701780756,
        "sunset_ts": 1701816504,
        "temp": 7.4,
        "ts": 1701756060,
        "uv": 3.2,
        "valid_date": "2023-12-05",
        "vis": 20.89,
        "weather": {
            "code": 801,
            "icon": "c02d",
            "description": "Few clouds"
        },
        "wind_cdir": "W",
        "wind_cdir_full": "west",
        "wind_dir": 266,
        "wind_gust_spd": 6.6,
        "wind_spd": 4.4
    },
    {
        "app_max_temp": 9,
        "app_min_temp": -2.4,
        "clouds": 7,
        "clouds_hi": 9,
        "clouds_low": 3,
        "clouds_mid": 0,
        "datetime": "2023-12-06",
        "dewpt": 1,
        "high_temp": 9.5,
        "low_temp": 2.6,
        "max_dhi": null,
        "max_temp": 9.5,
        "min_temp": 0.9,
        "moon_phase": 0.265408,
        "moon_phase_lunation": 0.8,
        "moonrise_ts": 1701844491,
        "moonset_ts": 1701891185,
        "ozone": 316,
        "pop": 0,
        "precip": 0,
        "pres": 1016.4,
        "rh": 75,
        "slp": 1026.5,
        "snow": 0,
        "snow_depth": 0,
        "sunrise_ts": 1701867206,
        "sunset_ts": 1701902906,
        "temp": 5.2,
        "ts": 1701842460,
        "uv": 3,
        "valid_date": "2023-12-06",
        "vis": 24.128,
        "weather": {
            "code": 801,
            "icon": "c02d",
            "description": "Few clouds"
        },
        "wind_cdir": "SW",
        "wind_cdir_full": "southwest",
        "wind_dir": 223,
        "wind_gust_spd": 4.8,
        "wind_spd": 3.1
    },
    {
        "app_max_temp": 12.8,
        "app_min_temp": -0.9,
        "clouds": 18,
        "clouds_hi": 42,
        "clouds_low": 0,
        "clouds_mid": 10,
        "datetime": "2023-12-07",
        "dewpt": 2.5,
        "high_temp": 13.4,
        "low_temp": 7.3,
        "max_dhi": null,
        "max_temp": 13.4,
        "min_temp": 2.6,
        "moon_phase": 0.182187,
        "moon_phase_lunation": 0.83,
        "moonrise_ts": 1701934276,
        "moonset_ts": 1701979057,
        "ozone": 291.3,
        "pop": 0,
        "precip": 0,
        "pres": 1010.7,
        "rh": 68,
        "slp": 1020.6,
        "snow": 0,
        "snow_depth": 0,
        "sunrise_ts": 1701953655,
        "sunset_ts": 1701989311,
        "temp": 8.2,
        "ts": 1701928860,
        "uv": 3,
        "valid_date": "2023-12-07",
        "vis": 24.128,
        "weather": {
            "code": 801,
            "icon": "c02d",
            "description": "Few clouds"
        },
        "wind_cdir": "S",
        "wind_cdir_full": "south",
        "wind_dir": 186,
        "wind_gust_spd": 8.5,
        "wind_spd": 5.7
    },
    {
        "app_max_temp": 19.4,
        "app_min_temp": 8.4,
        "clouds": 49,
        "clouds_hi": 47,
        "clouds_low": 52,
        "clouds_mid": 1,
        "datetime": "2023-12-08",
        "dewpt": 8.7,
        "high_temp": 20,
        "low_temp": 14.3,
        "max_dhi": null,
        "max_temp": 20,
        "min_temp": 7.3,
        "moon_phase": 0.109857,
        "moon_phase_lunation": 0.87,
        "moonrise_ts": 1702024143,
        "moonset_ts": 1702067105,
        "ozone": 282.9,
        "pop": 20,
        "precip": 0.125,
        "pres": 1005.3,
        "rh": 73,
        "slp": 1015.2,
        "snow": 0,
        "snow_depth": 0,
        "sunrise_ts": 1702040103,
        "sunset_ts": 1702075718,
        "temp": 13.6,
        "ts": 1702015260,
        "uv": 3.3,
        "valid_date": "2023-12-08",
        "vis": 23.4,
        "weather": {
            "code": 803,
            "icon": "c03d",
            "description": "Broken clouds"
        },
        "wind_cdir": "S",
        "wind_cdir_full": "south",
        "wind_dir": 186,
        "wind_gust_spd": 8.2,
        "wind_spd": 6.1
    },
    {
        "app_max_temp": 15.4,
        "app_min_temp": 8.9,
        "clouds": 64,
        "clouds_hi": 14,
        "clouds_low": 64,
        "clouds_mid": 1,
        "datetime": "2023-12-09",
        "dewpt": 9.4,
        "high_temp": 15.2,
        "low_temp": 3.4,
        "max_dhi": null,
        "max_temp": 15.4,
        "min_temp": 8.9,
        "moon_phase": 0.0526869,
        "moon_phase_lunation": 0.9,
        "moonrise_ts": 1702114163,
        "moonset_ts": 1702155437,
        "ozone": 301.2,
        "pop": 70,
        "precip": 4.5625,
        "pres": 1003.3,
        "rh": 77,
        "slp": 1013,
        "snow": 0,
        "snow_depth": 0,
        "sunrise_ts": 1702126550,
        "sunset_ts": 1702162127,
        "temp": 13.3,
        "ts": 1702101660,
        "uv": 3.4,
        "valid_date": "2023-12-09",
        "vis": 22.392,
        "weather": {
            "code": 520,
            "icon": "r04d",
            "description": "Light shower rain"
        },
        "wind_cdir": "SSW",
        "wind_cdir_full": "south-southwest",
        "wind_dir": 195,
        "wind_gust_spd": 4.6,
        "wind_spd": 4.6
    },
    {
        "app_max_temp": 8.4,
        "app_min_temp": -0.5,
        "clouds": 2,
        "clouds_hi": 0,
        "clouds_low": 2,
        "clouds_mid": 0,
        "datetime": "2023-12-10",
        "dewpt": 1.4,
        "high_temp": 8.4,
        "low_temp": 2.7,
        "max_dhi": null,
        "max_temp": 8.4,
        "min_temp": 3.4,
        "moon_phase": 0.0151441,
        "moon_phase_lunation": 0.93,
        "moonrise_ts": 1702204385,
        "moonset_ts": 1702244182,
        "ozone": 353.1,
        "pop": 0,
        "precip": 0,
        "pres": 1010.8,
        "rh": 75,
        "slp": 1020.8,
        "snow": 0,
        "snow_depth": 0,
        "sunrise_ts": 1702212996,
        "sunset_ts": 1702248538,
        "temp": 5.6,
        "ts": 1702188060,
        "uv": 3.4,
        "valid_date": "2023-12-10",
        "vis": 24.128,
        "weather": {
            "code": 801,
            "icon": "c02d",
            "description": "Few clouds"
        },
        "wind_cdir": "NW",
        "wind_cdir_full": "northwest",
        "wind_dir": 315,
        "wind_gust_spd": 4.2,
        "wind_spd": 4.2
    }
]