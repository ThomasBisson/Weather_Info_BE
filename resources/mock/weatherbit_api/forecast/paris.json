[
	{
		"app_max_temp":-0.4,
		"app_min_temp":-7.1,
		"clouds":69,
		"clouds_hi":52,
		"clouds_low":45,
		"clouds_mid":77,
		"datetime":"2023-12-03",
		"dewpt":-3.4,
		"high_temp":3.6,
		"low_temp":0.9,
		"max_dhi":null,
		"max_temp":3.6,
		"min_temp":-1.8,
		"moon_phase":0.568197,
		"moon_phase_lunation":0.7,
		"moonrise_ts":1701639800,
		"moonset_ts":1701606749,
		"ozone":295.4,
		"pop":20,
		"precip":0.016601562,
		"pres":1010.4,
		"rh":72,
		"slp":1017.9,
		"snow":0.34230864,
		"snow_depth":0,
		"sunrise_ts":1701588254,
		"sunset_ts":1701618863,
		"temp":1.2,
		"ts":1701558060,
		"uv":0,
		"valid_date":"2023-12-03",
		"vis":24.128,
		"weather":{
			"code":623,
			"icon":"s06d",
			"description":"Flurries"
		},
		"wind_cdir":"SSE",
		"wind_cdir_full":"south-southeast",
		"wind_dir":165,
		"wind_gust_spd":7.5,
		"wind_spd":3.6
	},
	{
		"app_max_temp":7.9,
		"app_min_temp":-2.5,
		"clouds":92,
		"clouds_hi":55,
		"clouds_low":75,
		"clouds_mid":93,
		"datetime":"2023-12-04",
		"dewpt":2.3,
		"high_temp":7.9,
		"low_temp":6.3,
		"max_dhi":null,
		"max_temp":7.9,
		"min_temp":2,
		"moon_phase":0.47354,
		"moon_phase_lunation":0.73,
		"moonrise_ts":1701730326,
		"moonset_ts":1701693987,
		"ozone":301.6,
		"pop":80,
		"precip":7.4,
		"pres":995.4,
		"rh":84,
		"slp":1002.6,
		"snow":0,
		"snow_depth":0,
		"sunrise_ts":1701674728,
		"sunset_ts":1701705241,
		"temp":4.8,
		"ts":1701644460,
		"uv":0.7,
		"valid_date":"2023-12-04",
		"vis":19.703,
		"weather":{
			"code":610,
			"icon":"s04d",
			"description":"Mix snow/rain"
		},
		"wind_cdir":"S",
		"wind_cdir_full":"south",
		"wind_dir":181,
		"wind_gust_spd":8.1,
		"wind_spd":4.9
	},
	{
		"app_max_temp":3.4,
		"app_min_temp":1.7,
		"clouds":93,
		"clouds_hi":11,
		"clouds_low":86,
		"clouds_mid":86,
		"datetime":"2023-12-05",
		"dewpt":4.5,
		"high_temp":6.3,
		"low_temp":3.2,
		"max_dhi":null,
		"max_temp":6.8,
		"min_temp":4.6,
		"moon_phase":0.379401,
		"moon_phase_lunation":0.76,
		"moonrise_ts":1701734377,
		"moonset_ts":1701781146,
		"ozone":302,
		"pop":75,
		"precip":5.7000003,
		"pres":999.7,
		"rh":90,
		"slp":1006.9,
		"snow":0,
		"snow_depth":0,
		"sunrise_ts":1701761199,
		"sunset_ts":1701791622,
		"temp":6.1,
		"ts":1701730860,
		"uv":1.1,
		"valid_date":"2023-12-05",
		"vis":23.639,
		"weather":{
			"code":500,
			"icon":"r01d",
			"description":"Light rain"
		},
		"wind_cdir":"WSW",
		"wind_cdir_full":"west-southwest",
		"wind_dir":258,
		"wind_gust_spd":7.6,
		"wind_spd":5.5
	},
	{
		"app_max_temp":3.1,
		"app_min_temp":0.3,
		"clouds":76,
		"clouds_hi":15,
		"clouds_low":59,
		"clouds_mid":18,
		"datetime":"2023-12-06",
		"dewpt":2.7,
		"high_temp":4.9,
		"low_temp":2.1,
		"max_dhi":null,
		"max_temp":4.9,
		"min_temp":2.8,
		"moon_phase":0.288416,
		"moon_phase_lunation":0.8,
		"moonrise_ts":1701820777,
		"moonset_ts":1701868291,
		"ozone":288.1,
		"pop":0,
		"precip":0,
		"pres":1009.4,
		"rh":92,
		"slp":1016.2,
		"snow":0,
		"snow_depth":0,
		"sunrise_ts":1701847669,
		"sunset_ts":1701878006,
		"temp":3.9,
		"ts":1701817260,
		"uv":1.8,
		"valid_date":"2023-12-06",
		"vis":24.128,
		"weather":{
			"code":804,
			"icon":"c04d",
			"description":"Overcast clouds"
		},
		"wind_cdir":"SW",
		"wind_cdir_full":"southwest",
		"wind_dir":227,
		"wind_gust_spd":3.1,
		"wind_spd":2.3
	},
	{
		"app_max_temp":7.3,
		"app_min_temp":-0.6,
		"clouds":85,
		"clouds_hi":93,
		"clouds_low":41,
		"clouds_mid":76,
		"datetime":"2023-12-07",
		"dewpt":1.9,
		"high_temp":8,
		"low_temp":7,
		"max_dhi":null,
		"max_temp":7.3,
		"min_temp":2.1,
		"moon_phase":0.203596,
		"moon_phase_lunation":0.83,
		"moonrise_ts":1701911218,
		"moonset_ts":1701955485,
		"ozone":275.7,
		"pop":55,
		"precip":2.4,
		"pres":999.3,
		"rh":82,
		"slp":1003.5,
		"snow":0,
		"snow_depth":0,
		"sunrise_ts":1701934137,
		"sunset_ts":1701964394,
		"temp":4.8,
		"ts":1701903660,
		"uv":0.6,
		"valid_date":"2023-12-07",
		"vis":20.581,
		"weather":{
			"code":520,
			"icon":"r04d",
			"description":"Light shower rain"
		},
		"wind_cdir":"SSE",
		"wind_cdir_full":"south-southeast",
		"wind_dir":156,
		"wind_gust_spd":4.7,
		"wind_spd":3.4
	},
	{
		"app_max_temp":8.9,
		"app_min_temp":4.5,
		"clouds":92,
		"clouds_hi":69,
		"clouds_low":32,
		"clouds_mid":45,
		"datetime":"2023-12-08",
		"dewpt":6.1,
		"high_temp":8.9,
		"low_temp":6.4,
		"max_dhi":null,
		"max_temp":8.9,
		"min_temp":7.1,
		"moon_phase":0.128476,
		"moon_phase_lunation":0.87,
		"moonrise_ts":1702001736,
		"moonset_ts":1702042802,
		"ozone":343.5,
		"pop":65,
		"precip":4,
		"pres":990.4,
		"rh":88,
		"slp":994.5,
		"snow":0,
		"snow_depth":0,
		"sunrise_ts":1702020603,
		"sunset_ts":1702050784,
		"temp":8,
		"ts":1701990060,
		"uv":2.1,
		"valid_date":"2023-12-08",
		"vis":24.128,
		"weather":{
			"code":520,
			"icon":"r04d",
			"description":"Light shower rain"
		},
		"wind_cdir":"S",
		"wind_cdir_full":"south",
		"wind_dir":188,
		"wind_gust_spd":6.8,
		"wind_spd":3.9
	},
	{
		"app_max_temp":8.4,
		"app_min_temp":4.5,
		"clouds":88,
		"clouds_hi":22,
		"clouds_low":71,
		"clouds_mid":44,
		"datetime":"2023-12-09",
		"dewpt":6.4,
		"high_temp":8.7,
		"low_temp":7.4,
		"max_dhi":null,
		"max_temp":8.4,
		"min_temp":6.4,
		"moon_phase":0.0671024,
		"moon_phase_lunation":0.9,
		"moonrise_ts":1702092416,
		"moonset_ts":1702130345,
		"ozone":320,
		"pop":0,
		"precip":0,
		"pres":994.1,
		"rh":92,
		"slp":998.2,
		"snow":0,
		"snow_depth":0,
		"sunrise_ts":1702107066,
		"sunset_ts":1702137177,
		"temp":7.6,
		"ts":1702076460,
		"uv":2.1,
		"valid_date":"2023-12-09",
		"vis":24.128,
		"weather":{
			"code":804,
			"icon":"c04d",
			"description":"Overcast clouds"
		},
		"wind_cdir":"S",
		"wind_cdir_full":"south",
		"wind_dir":186,
		"wind_gust_spd":5.1,
		"wind_spd":2.8
	},
	{
		"app_max_temp":10.1,
		"app_min_temp":7.4,
		"clouds":87,
		"clouds_hi":13,
		"clouds_low":32,
		"clouds_mid":5,
		"datetime":"2023-12-10",
		"dewpt":7.7,
		"high_temp":10.1,
		"low_temp":8.4,
		"max_dhi":null,
		"max_temp":10.1,
		"min_temp":7.4,
		"moon_phase":0.0238018,
		"moon_phase_lunation":0.93,
		"moonrise_ts":1702183320,
		"moonset_ts":1702218270,
		"ozone":273.8,
		"pop":65,
		"precip":3.6,
		"pres":995.8,
		"rh":93,
		"slp":999.8,
		"snow":0,
		"snow_depth":0,
		"sunrise_ts":1702193528,
		"sunset_ts":1702223574,
		"temp":8.8,
		"ts":1702162860,
		"uv":2.1,
		"valid_date":"2023-12-10",
		"vis":23.254,
		"weather":{
			"code":520,
			"icon":"r04d",
			"description":"Light shower rain"
		},
		"wind_cdir":"SSW",
		"wind_cdir_full":"south-southwest",
		"wind_dir":201,
		"wind_gust_spd":5.2,
		"wind_spd":3.8
	}
]