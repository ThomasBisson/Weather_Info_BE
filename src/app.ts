import express, { Application, Request, Response } from 'express';
import cors from 'cors';
import helmet from 'helmet';

import swagger from './swagger';

import errorHandling from './middlewares/errorHandling.middleware';
import morganMiddleware from './middlewares/morgan.middleware';

import weatherRouter from './routes/weather.router';

const app: Application = express();

app.use(morganMiddleware);
app.use(helmet());
app.use(cors());

// NOTE : Use express.json() or body-parser if you need to handle data in the request body
//        For now it's not needed so I commented as it slows the API for no good reason
// App.use(express.json());

// Swagger
if (process.env.NODE_ENV === 'development') {
	swagger(app);
}

// Routes
app.get('/', (req: Request, res: Response) => {
	res.json({ response: 'Welcome to the weather info API !' });
});

app.use('/weather', weatherRouter);

// 404 route
app.get('*', (req, res) => {
	res.status(404).json({
		error: `Cannot find ${req.method} ${req.path}`,
	});
});

// Error handler
app.use(errorHandling);

export default app;
