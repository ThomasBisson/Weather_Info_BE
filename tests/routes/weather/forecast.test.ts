import { describe, expect, jest, test } from '@jest/globals';
import request from 'supertest';

import app from './../../../src/app';

import * as forecastController from '../../../src/controllers/weather/forecast.controller';

import { WeatherbitForecastData } from '../../../src/types/weatherbit.type';

import { readMockJsonFile } from '../../../src/utils/file.util';

const spyGetWeatherbitData = jest.spyOn(forecastController, 'getWeatherbitData');

describe('Integration /weather/forecast', () => {
	test('return status 200 and result is OK for paris', async () => {
		const city = 'paris';
		const data = readMockJsonFile(city, 'forecast') as WeatherbitForecastData[];
		spyGetWeatherbitData.mockReturnValue(Promise.resolve(data));

		const res = await request(app).get(`/weather/forecast?city=${city}`);

		expect(res.statusCode).toBe(200);

		expect(res.body).toMatchSnapshot();
	});

	test('return status 200 and result is OK for toulouse', async () => {
		const city = 'toulouse';
		const data = readMockJsonFile(city, 'forecast') as WeatherbitForecastData[];
		spyGetWeatherbitData.mockReturnValue(Promise.resolve(data));

		const res = await request(app).get(`/weather/forecast?city=${city}`);

		expect(res.statusCode).toBe(200);

		expect(res.body).toMatchSnapshot();
	});

	test('return status 200 and result is OK for memphis', async () => {
		const city = 'memphis';
		const data = readMockJsonFile(city, 'forecast') as WeatherbitForecastData[];
		spyGetWeatherbitData.mockReturnValue(Promise.resolve(data));

		const res = await request(app).get(`/weather/forecast?city=${city}`);

		expect(res.statusCode).toBe(200);

		expect(res.body).toMatchSnapshot();
	});

	test('return status 200 and result is OK for sydney', async () => {
		const city = 'sydney';
		const data = readMockJsonFile(city, 'forecast') as WeatherbitForecastData[];
		spyGetWeatherbitData.mockReturnValue(Promise.resolve(data));

		const res = await request(app).get(`/weather/forecast?city=${city}`);

		expect(res.statusCode).toBe(200);

		expect(res.body).toMatchSnapshot();
	});
});
