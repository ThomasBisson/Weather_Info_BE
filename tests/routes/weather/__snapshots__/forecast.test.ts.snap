// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration /weather/forecast return status 200 and result is OK for memphis 1`] = `
{
  "avgWindBeaufortScale": "Gentle Breeze",
  "generalTrend": "stable",
  "pressureTrend": "stable",
  "temperatureTrend": "stable",
}
`;

exports[`Integration /weather/forecast return status 200 and result is OK for paris 1`] = `
{
  "avgWindBeaufortScale": "Gentle Breeze",
  "generalTrend": "improving",
  "pressureTrend": "falling",
  "temperatureTrend": "rising",
}
`;

exports[`Integration /weather/forecast return status 200 and result is OK for sydney 1`] = `
{
  "avgWindBeaufortScale": "Light Breeze",
  "generalTrend": "stable",
  "pressureTrend": "rising",
  "temperatureTrend": "stable",
}
`;

exports[`Integration /weather/forecast return status 200 and result is OK for toulouse 1`] = `
{
  "avgWindBeaufortScale": "Gentle Breeze",
  "generalTrend": "improving",
  "pressureTrend": "falling",
  "temperatureTrend": "rising",
}
`;
